apps/demos/src/app/rx-angular-pocs/template/directives/for/rx-for.directive.ts

```diff
@@ -22,6 +22,175 @@ import {
+ * templates based on a list of items.
+ * Input values can be provided either as `Observable` or `static` values. Just as the `*ngFor` directive, the
+ * `*rxFor` is placed on an
+ * element, which becomes the parent of the cloned templates.
+ *
+ * The `RxFor` implements `EmbeddedView Rendering`.
+ * Compared to the `NgForOf`, `RxFor` treats each child template as single renderable entity. For each
+ * change in the provided list of items it will apply and detect changes to only affected views.
+ *
+ * Under the hood, it leverages the power of the `StrategyCredential`s which in turn take care of scheduling and
+ * prioritizing the change detection for each child template (aka item in the list).
+ * This way the rendering behavior of each instance of `RxFor` can be configured individually.
+ *
+ * `StrategyCredentials` and `EmbeddedView Rendering` together build the basis for the `concurrent mode`. Based on
+ * the configured strategy every template will get processed in an individual task.
+ *
+ * As further improvement compared to the basic `*ngFor` implementation, `*rxFor` is able to take care of
+ * `ChangeDetection` for `projected views` (aka `@ContentChild` or `@ViewChild`).
+ *
+ * Read more about this in the example section
+ *
+ * ### Context Variables
+ *
+ * The following context variables are available for each template:
+ *
+ * - $implicit: `T`
+ * - item$: `Observable<T>`
+ * - index: `number`
+ * - count: `number`
+ * - first: `boolean`
+ * - last: `boolean`
+ * - even: `boolean`
+ * - odd: `boolean`
+ * - index$: `Observable<number>`
+ * - count$: `Observable<number>`
+ * - first$: `Observable<boolean>`
+ * - last$: `Observable<boolean>`
+ * - even$: `Observable<boolean>`
+ * - odd$: `Observable<boolean>`
+ * - select: (keys: string[], distinctByMap): Observable<Partial<T>> (reactivity in the embedded view)
+ *
+ * ### Input properties
+ *
+ *  - trackBy: `(index: number, item: T) => any`
+ *  - trackBy: `keyof T`
+ *  - distinctBy: `(item: T, item: T) => boolean`
+ *  - strategy: `string`
+ *  - strategy: `Observable<string>`
+ *  - parent: boolean;
+ *  - renderCallback: Subject<T[]>
+ *
+ *
+ * ### Features of `*rxFor`
+ *
+ * Included features for `*rxFor`:
+ * - Push based architecture
+ * - Immutable as well as mutable data structures (with `trackBy` & `distinctBy`)
+ * - Control `updates` via `distinctBy` and `moves` via `trackBy`
+ * - Reactive as well as imperative values in the template (`ngFor` drop-in replacement)
+ * - `ListManager`: special logic for differ mechanism to avoid over-rendering; abstracts away low level logic
+ * - render every EmbeddedView on its own while applying the configured StrategyCredentials#behavior
+ * - detach EmbeddedView to avoid ChangeDetection top-down rendering interference when scheduling
+ * - cancel any scheduled work if a remove was triggered for an `trackById` / `distinctById`
+ * - cancel an update if a new update was triggered for the same `trackById` / `distinctById`
```

Comment:
IMHO the `trackById` should not include `Id`?

Reply:
you are right regarding `distinctById` since this is a function which returns a boolean. `trackById` is pretty good imho since what comes out of `trackBy` is an identifier for each item in the list